modin/backends/base/query_compiler.py

```diff
@@ -1569,49 +1585,77 @@ def has_multiindex(self, axis=0):
         assert axis == 1
         return isinstance(self.columns, pandas.MultiIndex)
 
-    def get_index_name(self):
+    def get_index_name(self, axis=0):
         """
-        Get index name.
+        Get index name of specified axis.
+
+        Parameters
+        ----------
+        axis: int (default 0),
+            Axis to return index name on.
 
         Returns
         -------
         hashable
             Index name, None for MultiIndex.
         """
-        return self.index.name
+        return self.get_axis(axis).name
 
-    def set_index_name(self, name):
+    def set_index_name(
+        self, name=no_default, axis=0, index=no_default, columns=no_default
+    ):
         """
-        Set index name.
+        Set index name for the specified axis.
 
         Parameters
         ----------
-        name: hashable
+        keywords for axes: hashable,
```

Comment:
Please update docstring with all parameters.

Reply:
Describing all parameters and conditions of how they are used will make docstring too complicated, pandas just uses "keywords for axes" in those cases: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.reindex.html

can we keep it as it is?

Reply:
Maybe we can just simplify this function?

Reply:
simplified this function, now it has signature like this:
```python
def set_index_name(self, name, axis=0):
    ...
```
also added all parameters to the doc-string